numpy/random/_generator.pyx

```diff
@@ -4255,56 +4431,6 @@ cdef class Generator:
                     j = random_interval(&self._bitgen, i)
                     x[i], x[j] = x[j], x[i]
 
-    cdef inline _shuffle_raw(self, np.npy_intp n, np.npy_intp first,
```

Comment:
Is there anything gained by moving these functions to be standalone?

Reply:
The only object from the class that is needed by `_shuffle_raw()` and `_shuffle_int()` is the bit generator, so to increase the separation of concerns in the code, I changed them from methods to functions that accept the bit generator as an argument.  Now it is clear from just the signature of these functions that they do not rely on any other data or methods in the class.  And they are both now `nogil` C functions.